### PR TITLE
feat: add student groups

### PR DESCRIPTION
**Pull request by danieldietzler**

### CODE REVIEW

File: server/src/main/java/de/uftos/services/StudentGroupService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public StudentGroup update(String id, StudentGroup group) {
+        group.setId(id);
+
+        return this.repository.save(group);
+    }
+
+    public void addStudents(String id, List<Student> students) {
+        this.repository.addStudentsToGroups(students.stream().map((student) -> new StudentsGroups(id, student.getId())).toList());
```
Comment (Lama64) [inline server/src/main/java/de/uftos/services/StudentGroupService.java:50]: Arguments of StudentsGroups constructor call switched.

**Review by Lama64 — CHANGES_REQUESTED** [review]
Confusing name and switched arguments

File: server/src/main/java/de/uftos/dto/StudentsGroups.java [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package de.uftos.dto;
+
+public record StudentsGroups(String studentId, String studentGroupId) {}
```
[the remark this answers is not in the document]
Comment (danieldietzler) [reply server/src/main/java/de/uftos/dto/StudentsGroups.java:3]: Yeah, I had a really hard time coming up with a better name. Any ideas?

File: server/src/main/java/de/uftos/services/StudentGroupService.java (hunk 1 above)
Comment (danieldietzler) [reply server/src/main/java/de/uftos/services/StudentGroupService.java:50]: Good call!

File: server/src/main/java/de/uftos/dto/StudentsGroups.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lama64) [reply server/src/main/java/de/uftos/dto/StudentsGroups.java:3]: Maybe sth. like StudentAndGroup or StudentInGroup depending on the use cases of the record.
Just adding a word between Student and Group describing their relation in the record.

**Review by Lama64 — APPROVED** [review]

**Pull request merged**